polkadot/primitives/src/vstaging/mod.rs

```diff
@@ -50,6 +51,52 @@ impl Default for ApprovalVotingParams {
 	}
 }
 
+/// Coretime configuration parameters
+#[derive(
+	RuntimeDebug,
+	Copy,
+	Clone,
+	PartialEq,
+	Encode,
+	Decode,
+	TypeInfo,
+	serde::Serialize,
+	serde::Deserialize,
+)]
+pub struct CoretimeParams<BlockNumber> {
+	/// How many cores are managed by the coretime chain.
+	pub cores: u32,
+	/// The max number of times a claim can time out in availability
+	pub max_availability_timeouts: u32,
+	/// The number of blocks a claim stays in the scheduler's claimqueue before getting cleared.
+	/// This number should go reasonably higher than the number of blocks in the async backing
+	/// lookahead.
+	pub ttl: BlockNumber,
+	/// The maximum queue size of the pay as you go module.
+	pub on_demand_queue_max_size: u32,
+	/// The target utilization of the spot price queue in percentages.
+	pub on_demand_target_queue_utilization: Perbill,
+	/// How quickly the fee rises in reaction to increased utilization.
+	/// The lower the number the slower the increase.
+	pub on_demand_fee_variability: Perbill,
+	/// The minimum amount needed to claim a slot in the spot pricing queue.
+	pub on_demand_base_fee: Balance,
+}
```

Comment:
@antonva could you please have a look at the params I've extracted. Something missing? Something not supposed to be here? Names that can be improved?

The rest is not ready for review.